### PR TITLE
Add function "HitGeometry" to Secondaries

### LINKED ISSUE #238: Function of Secondaries 'DidHitDetector'
Add a function to `Secondaries`, which checks whether a specific Geometry has been hit. Returns a bool.

### PR DESCRIPTION
**Pull request by Jean1995**
Fixes #238 

This adds a function which checks whether a particle track has a hit a certain geometry or not. 
We decided that particles where the propagation ended on the border of a geometry (i.e. the particle did not enter the geometry, but "touched" it) will be counted as a hit.

### CODE REVIEW

File: tests/Secondaries_TEST.cxx [hunk 1]
```diff
@@ -154,6 +154,47 @@ TEST(SecondaryVector, EntryPointExitPointRePropagation)
...
+    std::vector<Cartesian3D> positions{ {0, 0, 0}, {0, 0, 10}, {0, 0, 20} };
+
+    for (auto p : positions) {
+        ParticleState p_state;
+        p_state.position = p;
+        p_state.direction = Cartesian3D(0, 0, 1);
+        dummy_track.push_back(p_state, InteractionType::Undefined);
+    }
+
+    // test geometries along propagation axis
```
Comment (sudojan) [inline tests/Secondaries_TEST.cxx:170]: Some test spheres seem redundant, while I would also suggest testing the other geometries.
Would be interesting to see, if the edge case of propagating at the surface of a box is also handled as expected.

**Review by sudojan — CHANGES_REQUESTED** [review]
pybindings are missing

File: tests/Secondaries_TEST.cxx (hunk 1 above)
Comment (Jean1995) [reply tests/Secondaries_TEST.cxx:172]: I added some tests for other geometries, although I think that this is rather something that should be tested in the UnitTests of the corresponding geometries.
However, I still think that the border case that you described could potentially be pretty important.

**Comment by Jean1995** [comment]
> pybindings are missing

Done

**Review by sudojan — APPROVED** [review]
